packages/dev/core/src/Rendering/renderingGroup.ts

```diff
@@ -354,6 +337,25 @@ export class RenderingGroup {
         return 0;
     }
 
+    /**
+     * Build in function which can be applied to ensure meshes of a special queue (opaque, alpha test, transparent)
+     * are grouped by material then geometry.
+     *
+     * @param a The first submesh
+     * @param b The second submesh
+     * @returns The result of the comparison
+     */
+    public static PainterSortCompare(a: SubMesh, b: SubMesh): number {
+        const meshA = a.getMesh();
+        const meshB = b.getMesh();
+
+        if (meshA.material && meshB.material) {
+            return meshA.material.uniqueId - meshB.material.uniqueId;
+        }        
+
+        return meshA.uniqueId - meshB.uniqueId;

```

Comment:
very nitty, but - there is always a chance (very low one?)  that the delta between two meshes' uniqueId will be the same as two different meshes with materials. Would that be an issue here?

Reply:
Nope at the very worse we will miss the cache